docs/tutorial/streamer_design.md

```diff
@@ -1,29 +1,398 @@
+| spatial stride 0    |       4         |   RW    | Spatial stride for input A                           |
+| spatial stride 1    |       5         |   RW    | Spatial stride for input B                           |     
+| spatial stride 2    |       6         |   RW    | Spatial stride for output OUT                        |     
+| base addr 0         |       7         |   RW    | Base address for input A                             |
+| base addr 1         |       8         |   RW    | Base address for input A                             |
+| base addr 2         |       9         |   RW    | Base address for output OUT                          |
+| start               |       10        |   RW    | Start register to send configurations                |
+| perf. counter       |       11        |   RO    | Performance counter indicating number of cycles ran  | 
+
+
+# Example Streamer Generation
+
+This is a good time to test our wrapper generation and see the changes in the CSR manager. More details about the wrapper generation are in the [Connect The Shell](./connect_shell.md) section. Do the following (this assumes that you are running within the container):
+
+1 - Go to the root directory.
+
+2 - Run the command:
+
+```bash
+/repo/util/wrappergen/wrappergen.py --cfg_path="/repo/target/snitch_cluster/cfg/snax-alu.hjson" --tpl_path="/repo/hw/templates/" --chisel_path="/repo/hw/chisel/" --gen_path="/repo/target/snitch_cluster/generated/"
+```
+
+!!! note
+
+    If you are working outside of the container, note that the `/repo` pertains to the root of the repository.
+
+3 - Wait a while since this generates the CSR manager, streamer, and all other wrappers.
+
+4 - When finished, navigate to `/repo/target/snitch_cluster/generated/snax_alu/`
+
+5 - Open the file `snax_alu_streamer_StreamerTop.sv`. This is the Chisel-generated file. It looks like a synthesized netlist. All modules required for the streamer are declared in this file. 
+
+6 - Find the top module `snax_alu_streamer_StreamerTop` within the `snax_alu_streamer_StreamerTop.sv` file. Can you identify the signals discussed in this section?
+
+- Which are the interfaces for the accelerator side?
+- Can you identify which are for the write and read ports?
+- How many are for write ports and how many are for read ports?
+- Which are the interfaces for the TCDM side? How many TCDM ports are there?
+
+7 - In the same `snax_alu_streamer_StreamerTop.sv` file find the `snax_alu_streamer_CsrManager` module. This is the CSR manager of the streamer.
+
+- Can you verify and count how many RW and RO ports there are?
+- Why do you think the start register is missing in the listed ports?
+- Which CSR number do you think pertains to the spatial stride for input A?
+
+7 - Find the generated wrapper `snax_alu_streamer_wrapper.sv`. 
+
+ - Can you see where the Chisel-generated streamer is instanced?
+ - Can you tell what the SNAX streamer wrapper is trying to fix?
```

Comment:
Maybe add answers to the questions in
```markdown
<details><summary>solution</summary> This is the solution! </details>
```
Rendered:
<details><summary>solution</summary> This is the solution! </details>

Reply:
Yes! Hold-on hehe I have to get the answers too 😅